### ISSUE TITLE
image pull secrets for flux

### ISSUE THREAD

**Issue by kwontae** [fluxcd/flux#2564]
**Describe the bug**
Even though I added the imagePullSecrets, flux is have unauthorized issue with my private image registry when scanning for running pod image tags.

**To Reproduce**
Steps to reproduce the behaviour:
0. What's your setup?
1. I've created a flux.yaml file using fluxct install.
2. Changed the flux image repository to my private azure container registry.
3. Add image pull secrets and policy
```
imagePullSecrets:
- name: flux
conatiners:
- name: flux
  image: example.azurecr.io/fluxcd/flux:tag
  imagePullPolicy: Always
...
...
```

**Expected behavior**
When flux scans the pods and look for their image tags to my ACR, it should be able to find it and update accordingly. However, it's having an unauthorized error.

**Logs**
If applicable, please provide logs of `fluxd` or the helm-operator. In a standard stand-alone installation of Flux, you'd get this by running `kubectl logs -n default deploy/flux`.
```
err="requesting tags: Get https://example.azurecr.io/v2/exampleImage/tags/list: unauthorized: authentication required"
```
**Additional context**
Add any other context about the problem here, e.g 
- Flux version: 1.15.0
- Helm Operator version: N/A
- Kubernetes version: 
- Git provider: Github Enterprise
- Container registry provider: Azure Container Registry

**Comment by lgebhardt** [comment]
@kwontae I was running into this, and added a note to the issue I opened that might help you https://github.com/fluxcd/flux/issues/2589#issuecomment-552089581

**Comment by 2opremio** [comment]
I suspect this could have been caused by the same problem fixed by #2728 . @kwontae  could you test whether 2opremio/flux:scope-imagepullsecrets-correctly-a802915a fixes the problem?

**Comment by kwontae** [comment]
hey @2opremio! I don't know why github decided to send me a notification email about this now... Is your fix included in version 1.17.0? That's the current version i'm running for flux.

**Comment by 2opremio** [comment]
@kwontae the fix is in 1.17.1
